cmd/acr/purge.go

```diff
@@ -463,32 +467,46 @@ func getManifestsToDelete(ctx context.Context, acrClient api.AcrCLIClientInterfa
 		return nil, err
 	}
 	// This will act as a set if a key is present then it should not be deleted because it is referenced by a multiarch manifest
-	// that will not be deleted
-	doNotDelete := map[string]bool{}
-	candidatesToDelete := []acr.ManifestAttributesBase{}
-	// Iterate over all manifests to discover multiarchitecture manifests
+	// or the manifest has subjects attached
+	doNotDelete := make(map[string]struct{})
```

Comment:
it would be nicer to create a `Set` structure.

Reply:
Created a set.go and set_tests.go.  The idea is based on map.